projects/js-packages/publicize-components/src/store/selectors.js

```diff
@@ -570,3 +570,12 @@ export function getJetpackSocialPostAlreadyShared() {
 export function getAttachedMedia() {
 	return get( getJetpackSocialOptions(), [ 'attached_media' ], [] );
 }
+
+/**
+ * Get a list of all image generator settings for a post.
+ *
+ * @returns {Array} An array of image generator settings.
+ */
+export function getImageGeneratorPostSettings() {
+	return get( getJetpackSocialOptions(), [ 'image_generator_settings' ], [] );
```

Comment:
I've just realised that we're using lodash here and I think we're trying to move away from it. Perhaps we could start with this? Do something like:

```suggestion
	return getJetpackSocialOptions()?.[ 'image_generator_settings' ] ?? [];
```